apis/v1alpha2/httproute_types.go

```diff
@@ -641,6 +657,27 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// HTTPRequestRedirect defines configuration for the RequestRedirect filter.
+type HTTPRequestRedirect struct {
+	// Protocol is the protocol to be used in the value of the `Location`
+	// header in the response.
+	// When empty, the protocol of the request is used.
+	// +kubebuilder:validation:Enum=HTTP;HTTPS
+	Protocol *string `json:"protocol,omitempty"`
+	// Hostname is the hostname to be used in the value of the `Location`
+	// header in the response.
+	// When empty, the hostname of the request is used.
+	Hostname *string `json:"hostname,omitempty"`
```

Comment:
Nit: even for pointer fields, I think it's good to tag with `+optional` just to make the semantics really explicit for the reader.